test/hotspot/jtreg/runtime/cds/appcds/jcmd/JCmdTestStaticDump.java

```diff
@@ -0,0 +1,168 @@
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+/*
+ * @test
+ * @bug 8259070
+ * @summary Test jcmd to dump static shared archive.
+ * @requires vm.cds
+ * @library /test/lib /test/hotspot/jtreg/runtime/cds/appcds
+ * @modules jdk.jcmd/sun.tools.common:+open
+ * @compile ../test-classes/Hello.java JCmdTestDumpBase.java
+ * @build sun.hotspot.WhiteBox
+ * @build JCmdTestLingeredApp JCmdTestStaticDump
+ * @run driver jdk.test.lib.helpers.ClassFileInstaller sun.hotspot.WhiteBox
+ * @run main/othervm/timeout=480 -Xbootclasspath/a:. -XX:+UnlockDiagnosticVMOptions -XX:+WhiteBoxAPI JCmdTestStaticDump
+ */
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+import jdk.test.lib.apps.LingeredApp;
+import jdk.test.lib.dcmd.PidJcmdExecutor;
+import jdk.test.lib.process.OutputAnalyzer;
+
+import java.io.InputStreamReader;
+import java.io.BufferedReader;
+
+public class JCmdTestStaticDump extends JCmdTestDumpBase {
+
+    static final String STATIC_DUMP_FILE    = "mystatic";
+    static final String[] STATIC_MESSAGES   = {"JCmdTestLingeredApp source: shared objects file",
+                                               "LingeredApp source: shared objects file",
+                                               "Hello source: shared objects file"};
+
+    // Those two flags will not create a successful LingeredApp.
+    private static String[] noDumpFlags  =
+        {"-XX:+DumpSharedSpaces",
+         "-Xshare:dump"};
+    // Those flags will be excluded in static dumping,
+    // See src/java.base/share/classes/jdk/internal/misc/CDS.java
+    private static String[] excludeFlags = {
+         "-XX:DumpLoadedClassList=AnyFileName.classlist",
+         // this flag just dump archive, won't run app normally.
+         // "-XX:+DumpSharedSpaces",
+         "-XX:+DynamicDumpSharedSpaces",
+         "-XX:+RecordDynamicDumpInfo",
+         "-Xshare:on",
+         "-Xshare:auto",
+         "-XX:SharedClassListFile=non-exist.classlist",
+         "-XX:SharedArchiveFile=non-exist.jsa",
+         "-XX:ArchiveClassesAtExit=tmp.jsa",
+         "-XX:+UseSharedSpaces",
+         "-XX:+RequireSharedSpaces"};
+
+    // Times to dump cds against same process.
+    private static final int ITERATION_TIMES = 2;
+
+    static void test() throws Exception {
+        setIsStatic(true);
+        buildJars();
+        int  test_count = 1;
+        final boolean useBoot = true;
+        final boolean noBoot = !useBoot;
+        final boolean EXPECT_PASS = true;
+        final boolean EXPECT_FAIL = !EXPECT_PASS;
```

Comment:
These are also used in JCmdTestDynamicDump. Maybe move them to JCmdTestDumpBase?